modules/monitoring-example-alertmanager-webhook-authentication-settings.adoc

```diff
@@ -0,0 +1,53 @@
+// Module included in the following assemblies:
+//
+// * monitoring/configuring-the-monitoring-stack.adoc
+
+:_content-type: REFERENCE
+[id="example-alertmanager-webhook-authentication-settings_{context}"]
+= Example Alertmanager webhook authentication settings
+
+These examples show settings that configure Alertmanager in the `openshift-monitoring` namespace to authenticate with a webhook endpoint by using an API token that is configured in a `Secret` object.
+
+The following sample YAML shows a `Secret` object with authentication credentials to be referenced in the Alertmanager config map settings for the Cluster Monitoring Operator (CMO):
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+data: 
+  api_token: m8VjfmV2Cg== <1>
+metadata: 
+  name: alertmanager-credentials <2>
+  namespace: openshift-monitoring
+----
+<1> Replace this value with the API token used to authenticate to your webhook endpoint.
+<2> Reference this name in the `secrets` field in the CMO config map settings. 
+
+The following sample YAML shows CMO config map settings for Alertmanager.
+These settings reference the authentication credentials contained in the previous `Secret` object, which is mounted as a volume in the `alertmanager` container for the Alertmanager pods:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    alertmanagerMain:
+      enabled: true
+      secrets:
+      - alertmanager-credentials <1> 
+      receivers:
```

Comment:
alertmanagerMain does not have `receivers` setting, see:
https://github.com/openshift/cluster-monitoring-operator/blob/release-4.14/pkg/manifests/types.go#L69-L102
it should be in AlertmanagerConfig, not in cluster-monitoring-config configmap

Reply:
Thanks @juzhao. 

@simonpasquier - WDYT? Can you suggest what changes I need to make here?

Reply:
@juzhao is of course 100% correct that the CMO configmap has no `receivers` setting! My bad for missing it.

We need another snippet showing the Alertmanager configuration that will end up in the `alertmanager-main` secret. Also we have instructions for the Alertmanager config in the "Managing alerts" page, not in "Configuring the monitoring stack":
https://docs.openshift.com/container-platform/4.13/monitoring/managing-alerts.html#applying-custom-alertmanager-configuration_managing-alerts

Maybe the new instructions should go there too?

Reply:
> @juzhao is of course 100% correct that the CMO configmap has no `receivers` setting! My bad for missing it.
> 
> We need another snippet showing the Alertmanager configuration that will end up in the `alertmanager-main` secret. Also we have instructions for the Alertmanager config in the "Managing alerts" page, not in "Configuring the monitoring stack": https://docs.openshift.com/container-platform/4.13/monitoring/managing-alerts.html#applying-custom-alertmanager-configuration_managing-alerts
> 
> Maybe the new instructions should go there too?

@simonpasquier yes, we could add under the doc you mentioned
first, we need to create the secret under openshift-monitoring first
```
apiVersion: v1
kind: Secret
data: 
  api_token: m8VjfmV2Cg== <1>
metadata: 
  name: alertmanager-credentials <2>
  namespace: openshift-monitoring
```
then reference the secret in 
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: cluster-monitoring-config
  namespace: openshift-monitoring
data:
  config.yaml: |
    alertmanagerMain:
      secrets:
      - alertmanager-credentials
```
follow https://docs.openshift.com/container-platform/4.13/monitoring/managing-alerts.html#applying-custom-alertmanager-configuration_managing-alerts
to add webhook configuration in alertmanager-main secret, and it seems we need to add matchers for webhook under route.routes, otherwise there is info in alertmanager
`ts=2023-09-25T07:34:49.298Z caller=main.go:451 level=info component=configuration msg="skipping creation of receiver not referenced by any route" receiver=webhook
`
alertmanager-main secret details for referrence
```
global:
  resolve_timeout: 5m
inhibit_rules:
  - equal:
      - namespace
      - alertname
    source_matchers:
      - severity = critical
    target_matchers:
      - severity =~ warning|info
  - equal:
      - namespace
      - alertname
    source_matchers:
      - severity = warning
    target_matchers:
      - severity = info
  - equal:
      - namespace
    source_matchers:
      - alertname = InfoInhibitor
    target_matchers:
      - severity = info
receivers:
  - name: Default
  - name: Watchdog
  - name: Critical
  - name: 'null'
  - name: webhook
    webhook_configs:
      - url: "https://webhook.example.com/"
        http_config:
          authorization:
            type: apikey
            credentials_file: /etc/alertmanager/secrets/alertmanager-credentials/api_token
route:
  group_by:
    - namespace
  group_interval: 5m
  group_wait: 30s
  receiver: Default
  repeat_interval: 12h
  routes:
    - matchers:
        - alertname = Watchdog
      receiver: Watchdog
    - matchers:
        - alertname = InfoInhibitor
      receiver: 'null'
    - matchers:
        - severity = critical
      receiver: Critical
    - matchers:
        - alertname = Watchdog
      receiver: webhook
```